src/client/mod.rs

```diff
@@ -288,6 +288,21 @@ impl Client {
         }
     }
 
+    /// A helper method to start a `ChangeStream`. Used by watch methods in client, coll, and db.
+    pub(crate) async fn start_change_stream(
+        &self,
+        pipeline: impl IntoIterator<Item = Document>,
+        mut options: Option<ChangeStreamOptions>,
+        target: ChangeStreamTarget,
+    ) -> Result<ChangeStream> {
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+        resolve_options!(self, options, [read_concern, selection_criteria]);
```

Comment:
Hm, I actually think we need to invoke this particular line in the individual `watch` helpers themselves, since otherwise it will always resolve using the options specified on the `Client` rather than the `Database` or `Collection` that `watch` may have been called from. Once you move this line to those places, can you add a test verifying the `ReadConcern` and `SelectionCriteria` are inherited properly?

Reply:
fixed